crates/bevy_math/src/bounding/raytest2d.rs

```diff
@@ -0,0 +1,98 @@
+use super::{Aabb2d, BoundingCircle, IntersectsVolume};
+use crate::{primitives::Direction2d, Ray2d, Vec2};
+
+/// A raycast intersection test for 2D bounding volumes
+pub struct RayTest2d {
+    /// The ray for the test
+    pub ray: Ray2d,
+    /// The maximum time of impact of the ray
+    pub max: f32,
```

Comment:
It could be useful to explain what the time of impact means. This has caused some confusion for people in bevy_xpbd/bevy_rapier (Googling "time of impact" also doesn't give relevant results).

I would even change this to be a maximum distance, as the direction is always guaranteed to be normalized (thanks to `Direction2d`), so the "ray speed" is always 1 unit/second. A maximum distance feels more intuitive.